### PR TITLE
[IMP] core: complement comparison with empty str

### PR DESCRIPTION
**Pull request by kmagusiak**
Comparing a char field to "" or to False should result in the same result. This is the first step to make char fields always return strings even if there is a null in the database; just like we do for integers.

task-4013641

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/mail/models/mail_thread.py [hunk 1]
```diff
@@ -915,7 +915,7 @@ def _detect_is_bounce(self, message, message_dict):
         bounce_aliases = self.env['mail.alias.domain'].search([]).mapped('bounce_email')
         email_to_list = [
             email_normalize(e) or e
-            for e in (email_split(message_dict['to']) or [''])
+            for e in (email_split(message_dict['to']))
```
Comment (rco-odoo) [inline addons/mail/models/mail_thread.py:918]:
```suggestion
            for e in email_split(message_dict['to'])
```

File: addons/mail/models/mail_thread.py [hunk 2]
```diff
@@ -1034,7 +1034,7 @@ def _detect_write_to_catchall(self, msg_dict):
         else:
             catchall_aliases = self.env['mail.alias.domain'].search([]).mapped('catchall_email')
 
-        email_to_list = [email_normalize(e) or e for e in (email_split(msg_dict['to']) or [''])]
+        email_to_list = [email_normalize(e) or e for e in (email_split(msg_dict['to']))]
```
Comment (rco-odoo) [inline addons/mail/models/mail_thread.py:1037]:
```suggestion
        email_to_list = [email_normalize(e) or e for e in email_split(msg_dict['to'])]
```

**Review by rco-odoo — APPROVED** [review]
@robodoo delegate+

**Comment by kmagusiak** [comment]
@robodoo r+

**Pull request closed without merging**